server/routes/api.js

```diff
@@ -195,4 +247,40 @@ router.post('/reload', function(req, res) {
 	res.json(true);
 });
 
+router.post('/is-viewable', function(req, res){
+
+	const url = req.body.url;
+
+	fetch(url, {
+			method: 'head'
+		})
+		.then(function(response){
+			return response.headers.get('x-frame-options');
```

Comment:
I think this is a synchronous function, there's no need to have it go into another thenable as that is actually less performant.